var/spack/repos/builtin/packages/hiop/package.py

```diff
@@ -63,6 +63,52 @@ class Hiop(CMakePackage, CudaPackage, ROCmPackage):
     variant("ginkgo", default=False, description="Enable/disable ginkgo solver")
     variant("cusolver", default=False, description="Enable/disable cuSovler")
 
+    variant(
+        "full_optimizations",
+        default=False,
+        description="Enable/Disable optimizations and release type",
+    )
+    depends_on("ginkgo+full_optimizations", when="+ginkgo+full_optimizations")
+    depends_on("ginkgo~full_optimizations", when="+ginkgo~full_optimizations")
+
+    # force optimizations flag to use Release mode
+    conflicts(
+        "build_type=Debug",
+        when="+full_optimizations",
+        msg="Use Release mode when with optimizations",
+    )
+    conflicts(
+        "build_type=MinSizeRel",
+        when="+full_optimizations",
+        msg="Use Release mode when with optimizations",
+    )
+    conflicts(
+        "build_type=RelWithDebInfo",
+        when="+full_optimizations",
+        msg="Use Release mode when with optimizations",
+    )
+    # force no optimizations flag to use RelWithDebInfo mode
+    conflicts(
+        "build_type=Release",
+        when="~full_optimizations",
+        msg="Use RelWithDebInfo mode when without optimizations",
+    )
+    conflicts(
+        "build_type=Debug",
+        when="~full_optimizations",
+        msg="Use RelWithDebInfo mode when without optimizations",
+    )
+    conflicts(
+        "build_type=MinSizeRel",
+        when="~full_optimizations",
+        msg="Use RelWithDebInfo mode when without optimizations",
+    )
+
+    # force dependencies build types for optimizations
+    for pkg in ["raja", "umpire", "magma", "camp"]:
```

Comment:
I think we can add ginkgo here as @tcojean suggests, but we might have to add a little logic to also propagate the `full_optimizations` flag to ginkgo.

Reply:
Yes adding Ginkgo here should be enough. Also, as I stated in https://github.com/spack/spack/pull/32745#pullrequestreview-1115781215 it is not necessary to use the `full_optimizations` flag with Ginkgo for HiOp, this specifically targets some optimizations for some preconditioners which you are not using, but increases the compile time significantly.

Reply:
Ah, I didn't quite catch that. Thank you. Should be sufficient just to specify the build type then, and we can leave `full_optimizations` alone to speed up our Ginkgo compile time!